Don't repeat zero commands to inverters

shsms opened the pull request:
Closes #1259

shsms committed:
Remove unnecessary `super().__init__()` call

The class doesn't derive from another class, so this call is
unnecessary.

Signed-off-by: Sahas Subramanian <sahas.subramanian@proton.me>

cwasicki commented on src/frequenz/sdk/microgrid/_power_distributing/_component_managers/_battery_manager.py:
```diff
@@ -271,21 +274,31 @@ async def _distribute_power(
+            if not (
+                dist == Power.zero()
+                and self._last_set_power.get(inverter_id) == Power.zero()
+            ):
```
Would it be possible to clean the distribution items first so that this check isn't needed in two places that are quite far apart.

shsms replied:
True.  The problem is that the distribution is created in a different class that doesn't have access to whether previous commands were successful or not.  And I don't want to modify the distribution object after it is created and returned by the other class.

But maybe it is less weird to pass the last_set_power information to that class each time.  I'll try that.

matthias-wende-frequenz replied:
Good catch. That should be possible and safe. (as long as nobody starts using _set_distributed_power from another place)

```
if inverter_id in distribution.distribution:
    del distribution.distribution[inverter_id]
```

shsms replied:
I'll just move the log line to the place where the power is set, such that there are checks only in one place.

shsms replied:
Ok, ready for a look again

cwasicki reviewed (commented):
LGTM

ela-kotulska-frequenz commented on src/frequenz/sdk/microgrid/_power_distributing/_component_managers/_battery_manager.py:
```diff
@@ -652,6 +665,12 @@ async def _set_distributed_power(
+                power == Power.zero()
+                and self._last_set_power.get(inverter_id) == Power.zero()
+            )
         }
```
I thought actors set lower and upper power limit, to not interfere with lower-level actors. 
And if actor won't use battery, then it can set None (then lower level actor will have full control).

shsms replied:
Yes, that's when they're using the battery pool, etc.  And the `None` requests are handled by the power manager.  Then the power manager sends a request to the power distributor which is what this PR is changing.

The power distributor can't send None values to the API, it has to send zero values.

ela-kotulska-frequenz replied:
Ok thanks a lot!

shsms committed:
Move battery distribution log message to just before API calls

This makes the logs display exactly those components for which API
calls are made.

Signed-off-by: Sahas Subramanian <sahas.subramanian@proton.me>

shsms committed:
Don't repeat zero commands to inverters

Signed-off-by: Sahas Subramanian <sahas.subramanian@proton.me>

shsms committed:
Update release notes

Signed-off-by: Sahas Subramanian <sahas.subramanian@proton.me>

llucax reviewed (approved):
Just a pedantic comment: technically all classes implicitly inherit from `object`, so they have a parent :)
